### ISSUE TITLE
Improve code example in "Configuring and Running a Job" section

### ISSUE THREAD

**Issue by benas** [spring-projects/spring-batch#3888]
The default constrcutor of `DefaultBatchConfigurer` is [protected](https://github.com/spring-projects/spring-batch/blob/d8fc58338d3b059b67b5f777adc132d2564d7402/spring-batch-core/src/main/java/org/springframework/batch/core/configuration/annotation/DefaultBatchConfigurer.java#L68) and can't be used in a code example like the following:

```java
@Bean
public BatchConfigurer batchConfigurer() {
   return new DefaultBatchConfigurer();
}
```

The code example that shows how to provide a custom transaction manager via a `BatchConfigurer` in the [Configuring and Running a Job](https://docs.spring.io/spring-batch/docs/4.3.x/reference/html/job.html#javaConfig) uses the default constructor of `DefaultBatchConfigurer`. While this is correct as the snippet uses an anonymous extension class, it does not show which datasource is being used as it is implicitly autowired. It is better to use the constructor that takes a `DataSource` which is more explicit about the datasource being used:

```diff
@Bean
--public BatchConfigurer batchConfigurer() {
++public BatchConfigurer batchConfigurer(DataSource dataSource) {

--	return new DefaultBatchConfigurer() {
++	return new DefaultBatchConfigurer(dataSource) {
		@Override
		public PlatformTransactionManager getTransactionManager() {
			return new MyTransactionManager();
		}
	};
}
```

**Comment by rchandavaram** [comment]
@benas 
I would like to work on this, please let me know if it is ok.
Any documentation for first timers is appreciated.

-RC

**Comment by benas** [comment]
@rchandavaram Thank you for your PR! I planned to merge the fix in the upcoming release.

**Comment by rchandavaram** [comment]
@benas Thank you for approving my first PR!

**Comment by benas** [comment]
Resolved with #3890 .
